### PR TITLE
add order for polynomial interpolation, fixes #8762

### LINKED ISSUE #8762: DataArrayResample.interpolate and passing kwargs
### What is your issue?

The docs [here](https://docs.xarray.dev/en/stable/generated/xarray.core.resample.DataArrayResample.interpolate.html) state that it is possible to use `polynomial` interpolation when resampling. 

If you try this though:

```python
data_per_minute = ds.resample(
    valid_time="1min"
).interpolate("polynomial")
```

You get the following:

```
ValueError: order is required when method=polynomial
```

The docs [here](https://docs.xarray.dev/en/stable/generated/xarray.core.resample.DataArrayResample.interpolate.html) state that you need to pass in order, how is this possible?

If you then try:

```python
data_per_minute = ds.resample(
    valid_time="1min"
).interpolate(kind="polynomial", order=3)
```

You get:

```
TypeError: Resample.interpolate() got an unexpected keyword argument 'order'
```

How can the order be specified? Thank you in advance.

### PR DESCRIPTION
**Pull request by nkarasiak**
- [x] Closes #8762
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

### CODE REVIEW

File: xarray/core/resample.py [hunk 1]
```diff
@@ -168,17 +168,18 @@ def interpolate(self, kind: InterpOptions = "linear") -> T_Xarray:
...
         """
-        return self._interpolate(kind=kind)
+        return self._interpolate(kind=kind, **kwargs)
 
-    def _interpolate(self, kind="linear") -> T_Xarray:
+    def _interpolate(self, kind="linear", **kwargs) -> T_Xarray:
         """Apply scipy.interpolate.interp1d along resampling dimension."""
         obj = self._drop_coords()
         (grouper,) = self.groupers
+        kwargs.update({"bounds_error": False})
```
Comment (dcherian) [inline xarray/core/resample.py:177]:
```suggestion
        kwargs.setdefault({"bounds_error": False})
```

Now that a user can set `kwargs`, this seems more appropriate?

**Comment by dcherian** [comment]
Thanks a lot @nkarasiak this is a nice first PR.

Can you update the test here please for `kind="polynomial", order=1`? https://github.com/pydata/xarray/blob/ef709df9623332e0a86881ef032b788fe5c51213/xarray/tests/test_groupby.py#L2055

File: xarray/core/resample.py (hunk 1 above)
Comment (nkarasiak) [reply xarray/core/resample.py]: Hi @dcherian , just change to `kwargs.setdefault("bounds_error", False)`. 
I didn't know the `setdefault` method, thank you !

**Comment by nkarasiak** [comment]
Tests and implementation are ok for me👍 
A bug happenned in `CI / windows-latest py3.12 (pull_request) Failing after 19m` that doesn't seem to come from any of my little update.

File: xarray/tests/test_groupby.py [hunk 2]
```diff
@@ -2596,3 +2606,7 @@ def test_default_flox_method() -> None:
         assert kwargs["method"] == "cohorts"
     else:
         assert "method" not in kwargs
+
+
+if __name__ == "__main__":
+    TestDataArrayResample().test_upsample_interpolate()
```
Comment (dcherian) [inline xarray/tests/test_groupby.py:2612]:
```suggestion
```

File: doc/whats-new.rst [hunk 3]
```diff
@@ -41,6 +41,12 @@ Deprecations
 Bug fixes
 ~~~~~~~~~
 
+- Following `DataArrayResample.interpolate and passing kwargs
+  <https://github.com/pydata/xarray/issues/8762>`_ interpolate method now
+  supports kwargs such as order for producing the polynomial interpolation.
+ (:issue:`8762`). By
+  `Nicolas Karasiak <https://github.com/nkarasiak>`_.
```
Comment (dcherian) [inline doc/whats-new.rst:48]:
```suggestion
- Following :py:meth`DataArrayResample.interpolate` and`DatasetResample.interpolate` method now
  support aribtrary kwargs such as ``order`` for polynomial interpolation.
 (:issue:`8762`). By
  `Nicolas Karasiak <https://github.com/nkarasiak>`_.
```

**Comment by dcherian** [comment]
Nice thanks @nkarasiak welcome to Xarray!

**Pull request merged**